src/blocks/remote-data-empty-result/edit.tsx

```diff
@@ -0,0 +1,43 @@
+/**
+ * WordPress dependencies
+ */
+import { useBlockProps } from '@wordpress/block-editor';
+import { BlockEditProps } from '@wordpress/blocks';
+import { Placeholder } from '@wordpress/components';
+
+import { useRemoteDataContext } from '@/blocks/remote-data-container/hooks/useRemoteDataContext';
+import { __ } from '@/utils/i18n';
+
+import './editor.scss';
+
+export function Edit( props: BlockEditProps< RemoteDataEmptyResultBlockAttributes > ): JSX.Element {
+	const { context } = props;
+	const blockProps = useBlockProps();
+
+	const { remoteData } = useRemoteDataContext( context );
+
+	if ( ! remoteData?.results ) {
+		return (
+			<div { ...blockProps }>
+				<Placeholder
+					label={ __( 'Remote Data Empty Result' ) }
+					instructions={ __(
+						'This block only works when placed inside a remote data block when there are no results. This block will be ignored as currently configured.'
+					) }
+				/>
+			</div>
+		);
+	}
+
+	if ( remoteData?.results.length === 0 ) {
```

Comment:
Only render this block if the results are empty, otherwise it's not shown.

Reply:
Yip, I didn't mean it won't be rendered at all. It is always rendered but just with a default div when there isn't a need to show it. It'll be a placeholder, or the empty message otherwise. 

Hopefully that clears it up - I should have said only render the empty message instead of "only render this block".